src/main/java/book/shop/spring/boot/intro/service/UserServiceImpl.java

```diff
@@ -47,4 +50,12 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
     public void deleteById(Long id) {
         userRepository.deleteById(id);
     }
+
+    @Override
+    public User getAuthenticatedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String email = authentication.getName();
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
```

Comment:
add parameter to message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("User not found"));
```